sdk/communication/azure-communication-messages/src/main/java/com/azure/communication/messages/models/MessageTemplateItem.java

```diff
@@ -91,4 +78,121 @@ public String getLanguage() {
+        return jsonWriter.writeEndObject();
+    }
+
+    /**
+     * Reads an instance of MessageTemplateItem from the JsonReader.
+     *
+     * @param jsonReader The JsonReader being read.
+     * @return An instance of MessageTemplateItem if the JsonReader was pointing to an instance of it, or null if it was pointing to JSON null.
+     * @throws IllegalStateException If the deserialized JSON object was missing any required properties.
+     * @throws IOException If an error occurs while reading the MessageTemplateItem.
+     */
+    @Generated
+    public static MessageTemplateItem fromJson(JsonReader jsonReader) throws IOException {
+        return jsonReader.readObject(reader -> {
+            String discriminatorValue = null;
+            try (JsonReader readerToUse = reader.bufferObject()) {
+                // Prepare for reading
+                readerToUse.nextToken();
+                while (readerToUse.nextToken() != JsonToken.END_OBJECT) {
+                    String fieldName = readerToUse.getFieldName();
+                    readerToUse.nextToken();
+                    if ("kind".equals(fieldName)) {
+                        discriminatorValue = readerToUse.getString();
+                        break;
+                    } else {
+                        readerToUse.skipChildren();
+                    }
+                }
+                // Use the discriminator value to determine which subtype should be deserialized.
+                if ("whatsApp".equals(discriminatorValue)) {
+                    return WhatsAppMessageTemplateItem.fromJson(readerToUse.reset());
+                } else {
+                    return fromJsonKnownDiscriminator(readerToUse.reset());
+                }
+            }
+        });
+    }
+
+    @Generated
+    static MessageTemplateItem fromJsonKnownDiscriminator(JsonReader jsonReader) throws IOException {
+        return jsonReader.readObject(reader -> {
+            String name = null;
+            String language = null;
+            MessageTemplateStatus status = null;
+            CommunicationMessagesChannel kind = null;
+            while (reader.nextToken() != JsonToken.END_OBJECT) {
+                String fieldName = reader.getFieldName();
+                reader.nextToken();
+                if ("name".equals(fieldName)) {
+                    name = reader.getString();
+                } else if ("language".equals(fieldName)) {
+                    language = reader.getString();
+                } else if ("status".equals(fieldName)) {
+                    status = MessageTemplateStatus.fromString(reader.getString());
+                } else if ("kind".equals(fieldName)) {
+                    kind = CommunicationMessagesChannel.fromString(reader.getString());
+                } else {
+                    reader.skipChildren();
+                }
+            }
+            MessageTemplateItem deserializedMessageTemplateItem = new MessageTemplateItem(language, status);
```

Comment:
The class was customized to `abstract` by service.

Reply:
The method can be deleted with customizations, but I think a better option would be allowing the type to be instantiated as Jackson had to be handling this before (likely by creating anonymous implementations)